docs/pages/enroll-resources/desktop-access/dynamic-registration.mdx

```diff
@@ -0,0 +1,135 @@
+a Teleport role with permissions to manage `dynamic_windows_desktop` resources.
+
+In the following example, a role allows a user to perform all possible
+operations against `dynamic_windows_desktop` resources:
+
+```yaml
+allow:
+  rules:
+    - resources:
+        - dynamic_windows_desktop
+      verbs: [list, create, read, update, delete]
+```
+
+## Enabling dynamic registration
+
+To enable dynamic registration, include a `resources` section in your Windows Desktop
+Service configuration with a list of resource label selectors you'd like this
+service to monitor for registering:
+
+```yaml
+windows_desktop_service:
+  enabled: "yes"
+  resources:
+  - labels:
+      "*": "*"
+```
+
+You can use a wildcard selector to register all dynamic Windows desktop resources in the cluster
+on the Windows Desktop Service or provide a specific set of labels for a subset:
+
+```yaml
+resources:
+- labels:
+    "env": "prod"
+- labels:
+    "env": "test"
+```
+
+## Creating a dynamic_windows_desktop resource
+
+Configure Teleport to register a Windows desktop dynamically by creating an `dynamic_windows_desktop`
+resource. The following example configures Teleport for connecting to Windows desktop
+called `example` at `localhost:3089`.
+
+```yaml
+kind: dynamic_windows_desktop
+version: v1
+metadata:
+  name: example
+  description: "Example desktop"
+  labels:
+    env: test
+spec:
+  addr: localhost:3089
+```
+
+The user creating the dynamic Windows desktop needs to have a role with access to the
+Windows desktop labels and the `dynamic_windows_desktop` resource.  In this example role the user can only
+create and maintain dynamic Windows desktops labeled `env: test`.
+
+```yaml
+kind: role
+metadata:
+  name: example
+spec:
+  allow:
+    windows_desktop_labels:
+      env: test
+    rules:
+    - resources:
+      - dynamic_windows_desktop
+      verbs: [list, create, read, update, delete]
+version: v7
+```
+
+To create a dynamic Windows desktop resource, run:
+
+<Tabs>
+    <TabItem scope={["oss", "enterprise"]} label="Self-Hosted">
+
+        ```code
+        # Log in to your cluster with tsh so you can use tctl from your local machine.
+        # You can also run tctl on your Auth Service host without running "tsh login"
+        # first.
+        $ tsh login --proxy=teleport.example.com --user=myuser
+        $ tctl create dynamic_windows_desktop.yaml
+        ```
+
+    </TabItem>
+    <TabItem scope={["cloud"]} label="Teleport Enterprise (Cloud)">
+
+        ```code
+        # Log in to your Teleport cluster so you can use tctl remotely.
+        $ tsh login --proxy=mytenant.teleport.sh --user=myuser
+        $ tctl create dynamic_windows_desktop.yaml
+        ```
+
+    </TabItem>
+
+</Tabs>
+
+After the resource has been created, it will appear among the list of available
+Windows desktops (in `tsh desktops ls` or UI) as long as at least one Windows Desktop Service
```

Comment:
```suggestion
Windows desktops (in the web UI) as long as at least one Windows Desktop Service
```

There's no such thing as `tsh desktops ls` - we never built it because you can't connect to desktops with tsh.